assets/js/script.js

```diff
@@ -0,0 +1,182 @@
+// Assignment Code
+const generateBtn = document.querySelector("#generate");
+const lowerCaseLetters = "abcdefghijklmnopqrstuvwxyz";
+const upperCaseLetters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+const numbersCriteria = "0123456789";
+const specialCharCriteria = " !\"#$%&'()*+,-./:;<=>?@[]^_`{|}~";
+
+// generate the list containing the final password criteria
+function criteriaGenerator() {
+  // object containing the criteria questions confirmation
+  const confirmObj = {
+    lowercase: confirm("Would you like to include lowercase letters?"),
+
+    uppercase: confirm("Would you like to include uppercase letters?"),
+
+    numbers: confirm("Would you like to include numbers?"),
+
+    specialChar: confirm("Would you like to include special characters?"),
+  };
+  // initialize the list of criteria
+  let finalCriteria = [];
+
+  //iterate over the object keys and values
+  for (const [key, value] of Object.entries(confirmObj)) {
+    //if one value confirmed add the relevant criteria to the list
+    if (key === "lowercase" && value) {
+      finalCriteria.push(lowerCaseLetters);
+    } else if (key === "uppercase" && value) {
+      finalCriteria.push(upperCaseLetters);
+    } else if (key === "numbers" && value) {
+      finalCriteria.push(numbersCriteria);
+    } else if (key === "specialChar" && value) {
+      finalCriteria.push(specialCharCriteria);
+    }
+  }
+
+  return finalCriteria;
+}
+
+//random function
+function random(firstNumber, lastNumber) {
+  return Math.floor(Math.random() * lastNumber + firstNumber);
+}
+
+const getPasswordLength = () => {
+  // prompt the user with the pass length question
+  let passwordLength = prompt(
+    "Please enter a password length starting from 8 to 128"
+  );
+
+  // initialize loop variables
+  let confirmMe = false; //retry confirm variable
+  let counter = 0; // loop count variable
+
+  // loop to validate the pass length
+  while (
+    (passwordLength < 8 && counter < 5) ||
+    (passwordLength > 128 && counter < 5)
+  ) {
+    //increment counter
+    counter += 1;
+
+    //check if number of retry was exceeded, 5 is max
+    if (counter === 5) {
+      alert("Retry count exceeded! Generator Terminated!");
+
+      // return invalid length -1 to trigger stop generator execution later
+      return -1;
+    } else {
+      // ask user if retry or cancel
+      confirmMe = confirm("Incorrect password length! Try again or stop?");
+
+      if (confirmMe) {
+        // ask for the length
+        passwordLength = prompt(
+          "Please enter a password length starting from 8 to 128"
+        );
+      } else {
+        //break the loop if cancel
+        alert("Generator Session Terminated!");
+
+        // return invalid length -1 to trigger stop generator execution later
+        return -1;
+      }
+    }
+  }
+
```

Comment:
An interesting approach here with the Retry count. I don't think this was a requirement but it's good you added this in and good practice for future work.